### DISCUSSION TITLE
Windows arm64 support

### DISCUSSION THREAD

**Discussion by aloshka** [johnlindquist/kit#1511] [Errors]
Since snapdragon cpus will become increasingly popular, what features are supported / not supported. On winarm64 things like set_selected_text, delete script, etc don't work. Had to find out by opening source, is there a list of features that work/do not work?

The most important one is set_selected_text. Is there anyway at all to make it work? Is it because of a nodejs/win32 issue?

Freaking awesome tool, absolutely love it.

**Comment by johnlindquist** [comment]
@aloshka Can you try the preview build of Script Kit v3 to see if those are fixed for you on arm64. I swapped out some of the keyboard libraries which may have fixed them. Can you try the latest preview build and let me know if they're fixed for you? If not, I'll dig into it:

https://github.com/johnlindquist/kitapp/releases/tag/untagged-53426fbfc73a1dc3b115

And thanks for the kind words!

**Comment by aloshka** [comment]
Tried getting to the release, but getting a 404

[cid:48fd2349-bc6a-4f76-8836-5734ca2a7e1b]


________________________________

**Reply by johnlindquist** [reply]
Whoops, hadn't clicked publish, here you go:

https://github.com/johnlindquist/kitapp/releases/tag/v2.24.0

**Comment by aloshka** [comment]
Love the new layout and the ctrl+k stuff.

But no, still same exact error message.

________________________________

**Reply by johnlindquist** [reply]
@aloshka What's the error message that you're seeing? When exactly is it appearing? Any steps to reproduce would be incredibly helpful 💛

**Comment by johnlindquist** [comment]
@aloshka Can you try this build? `await setSelectedText("test")` is working for me on windows arm64 (and x64):

https://github.com/script-kit/app/releases/tag/v2.26.4

**Comment by aloshka** [comment]
This version works!! OMG thank you!!

In case you are wondering what my use case is..

  1.
Copy & pasting random data that gets cleaned up into perfect SQL statements
     *
paste varchar
     *
paste as bracket
     *
paste int list
  2.
Convert excel text/fields into perfectly formatted c# $"" replace statements
  3.
clean out publish folders for visual stuio
  4.
string utilities, json minimize, json expand, base64 encode/decode

I know most of all that is dev stuff, but damn, this impresses everyone. 1-2 otkeys and everything is perfect.

Thank you!!!!!! you are awesome, let me know if I can do anything for you, this app is life, and I've been adding to my library weekly if not daily.

Thank you thank you!!!
________________________________

**Reply by aloshka** [reply]
It seems that version 2.26.4 works, but 2.3 gives the same errors as any other version outside of 2.26.4. Am I missing something?

Sidenote, how can we sign up for v3 beta?

**Reply by johnlindquist** [reply]
@aloshka Just grab the latest release from here: https://github.com/script-kit/app/releases/tag/v3.3.31

**Reply by aloshka** [reply]
You are awesome!

**Comment by aloshka** [comment]
Hi! I've installed every pre-release version on mac arm64 windows and especially on my surface pro windows arm64 windows. 

I found out that the following:
- 2.26.12 and up - does not install lots of errors (install errors, crashes on install)
- 2.26.11 - Last working version (works with install and perfect)
- 2.24.4 - first version to work (anything below this version has similar errors to 2.26.12 above)

Last version I checked was 3.5.12. 

This only seems to apply to arm64 windows

Does anyone have any ideas what the change was that doesn't allow windows arm64 to install? The only release notes I see that was changed is pnpm which makes sense since it's part of the install. But installing my own "working" version, doesn't seem to make a difference. 

Any help would be super appreciated. Version 3+ has features I really like.

**Reply by johnlindquist** [reply]
Investigating... I wasn't aware it broke. My main windows machine is x64, so I don't check arm64 nearly as often.

**Comment by aloshka** [comment]
Anything I can do at all to help test?  Would a Azure VM that runs Arch64 Windows 11 help?

________________________________

**Reply by johnlindquist** [reply]
It's ok, I've got a VM running on parallels. I've found the issue, just working through it. Hopefully will have a release today

**Reply by johnlindquist** [reply]
@aloshka Can you try this build please?

https://github.com/script-kit/app/releases/tag/v3.5.24

**Comment by aloshka** [comment]
Thank you! Worked perfectly.

I didn't have node installed, but after installing it works.

Logs before node install:

Scriptkit has become insanely important to me. Thank you!


________________________________
